src/agents/workflows/subgraphs/step4_features/solution/problem_utils.py

```
@@ -0,0 +1,172 @@
+    pages = ocr_blocks.get("pages", []) if isinstance(ocr_blocks, dict) else getattr(ocr_blocks, "pages", [])
+    
+    problems_map = {}
+    last_num = None
+    
+    for page in pages:
+        blocks = page.get("blocks", []) if isinstance(page, dict) else getattr(page, "blocks", [])
+        problem_seen_norms = {} 
+        
+        for block in blocks:
+            text = _block_text(block)
+            if not text: continue
```

Ruff (0.14.14)

[error] Multiple statements on one line (colon)

(E701)